### PR TITLE
Mep12 newscalarformatter demo.py

### PR DESCRIPTION
**Pull request by ericmjl**

### CODE REVIEW

File: examples/pylab_examples/newscalarformatter_demo.py [hunk 1]
```diff
@@ -1,88 +1,92 @@
...
-plot(-x*1e5, -x*1e-4)
-gca().xaxis.set_major_formatter(OldScalarFormatter())
-gca().yaxis.set_major_formatter(OldScalarFormatter())
+x = np.arange(0, 1, .01)
+f = plt.figure(figsize=(6, 6))
+f.text(0.5, 0.975, 'The old formatter',
+       horizontalalignment='center', verticalalignment='top')
+plt.subplot(221)
+plt.plot(x * 1e5 + 1e10, x * 1e-10 + 1e-5)
+plt.gca().xaxis.set_major_formatter(OldScalarFormatter())
```
Comment (tacaswell) [inline examples/pylab_examples/newscalarformatter_demo.py]: Can you capture this as `ax = plt.gca()` and then use that for the rest of the example?

File: examples/pylab_examples/newscalarformatter_demo.py [hunk 2]
```diff
@@ -1,88 +1,92 @@
...
-subplot(223)
-plot(-x*1e5 - 1e10, -x*1e-5 - 1e-10)
-gca().xaxis.set_major_formatter(OldScalarFormatter())
-gca().yaxis.set_major_formatter(OldScalarFormatter())
-subplot(224)
-plot(-x*1e5, -x*1e-4)
-gca().xaxis.set_major_formatter(OldScalarFormatter())
-gca().yaxis.set_major_formatter(OldScalarFormatter())
+x = np.arange(0, 1, .01)
+f = plt.figure(figsize=(6, 6))
```
Comment (tacaswell) [inline examples/pylab_examples/newscalarformatter_demo.py]: Probably should do this as `fig, [[ax1, ax2], [ax3, ax4]] = plt.subplots(2, 2, figsize=(6, 6))` and then use the objects for the rest of the example

File: examples/pylab_examples/newscalarformatter_demo.py [hunk 3]
```diff
@@ -1,88 +1,95 @@
...
+ax2.plot(x * 1e5, x * 1e-4)
+ax2.xaxis.set_major_formatter(OldScalarFormatter())
+ax2.yaxis.set_major_formatter(OldScalarFormatter())
+
+ax3.plot(-x * 1e5 - 1e10, -x * 1e-5 - 1e-10)
+ax3.xaxis.set_major_formatter(OldScalarFormatter())
+ax3.yaxis.set_major_formatter(OldScalarFormatter())
+
+ax4.plot(-x * 1e5, -x * 1e-4)
+ax4 = plt.gca()
```
Comment (thisch) [inline examples/pylab_examples/newscalarformatter_demo.py]: Do you really need ``gca()`` here?

Comment (thisch) [inline examples/pylab_examples/newscalarformatter_demo.py]: I guess this is just a leftover from commit e35dc89

Comment (ericmjl) [inline examples/pylab_examples/newscalarformatter_demo.py]: Yes, my bad!

**Pull request merged**